### PR TITLE
Updated REST API documentation for menu and menuItem endpoints

### PR DESCRIPTION
**Pull request by EldanizAkbar**

### CODE REVIEW

File: src/main/java/org/esi/grable/menus/controller/MenuController.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import org.esi.grable.menus.service.MenuService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+
+@RestController
+@RequestMapping("/menus")
```
Comment (karlsoosalu) [inline src/main/java/org/esi/grable/menus/controller/MenuController.java]: "menus" in here, but "menu" in documentation and Rest file

File: RestClientFile.rest [hunk 2]
```diff
@@ -1,39 +1,80 @@
...
+### Create a Menu
+POST http://localhost:8080/restaurants/:id/menu
+Content-Type: application/json
+
+{
+  "lastUpdated": "2024-03-13T08:00:00"
+}
 
+### Delete Menu by ID
+DELETE http://localhost:8080/restaurants/:id/menu/:menuId
```
Comment (karlsoosalu) [inline RestClientFile.rest]: Make sure these endpoints work in the local env

File: src/main/java/org/esi/grable/menus/controller/MenuController.java (hunk 1 above)
Comment (EldanizAkbar) [reply src/main/java/org/esi/grable/menus/controller/MenuController.java]: done

File: RestClientFile.rest (hunk 2 above)
Comment (EldanizAkbar) [reply RestClientFile.rest]: I updated entities but still there is problem with rest api. So I pushed updates to branch for review if you can find the issue.

Comment (karlsoosalu) [reply RestClientFile.rest]: Added small changes, so it should be working now and you can continue

File: src/main/java/org/esi/grable/menus/service/MenuService.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import java.util.Optional;
+
+@Service
+public class MenuService {
+
+    @Autowired
+    private MenuRepository menuRepository;
+
+    @Autowired
+    private RestaurantRepository restaurantRepository;
```
Comment (karlsoosalu) [inline src/main/java/org/esi/grable/menus/service/MenuService.java:20]: Ideally, I think we shouldn't use other repositories like that. But instead, use web client to fetch it. Example in [here](https://github.com/M-Gharib/ESI-W4/blob/main/product-service/src/main/java/com/esi/productservice/products/service/ProductService.java#L58)

**Pull request merged**